modules/tensor_mechanics/src/actions/GlobalStrainAction.C

```diff
@@ -132,6 +134,16 @@ GlobalStrainAction::act()
                                       Utility::string_to_enum<FEFamily>("LAGRANGE")),
                                _block_ids.empty() ? nullptr : &_block_ids);
     }
+    if (_global_disp.size() > 0)
+      for (unsigned int i = 0; i < _global_disp.size(); ++i)
+      {
+        std::string aux_var_name = _global_disp[i];
+
+        _problem->addAuxVariable(aux_var_name,
+                                 FEType(Utility::string_to_enum<Order>("FIRST"),
+                                        Utility::string_to_enum<FEFamily>("LAGRANGE")),
+                                 _block_ids.empty() ? nullptr : &_block_ids);
```

Comment:
This is fine, but I would suggest checking the actual order and family of the displacement variables. For second order lagrange the Auxvariables should be second order as well.